File: presentation/src/main/java/co/edu/uniautonoma/posgradosapp/presentation/ui/ubicacion/UbicacionViewModel.java
package co.edu.uniautonoma.posgradosapp.presentation.ui.ubicacion;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import co.edu.uniautonoma.posgradosapp.presentation.R;
import co.edu.uniautonoma.posgradosapp.domain.entity.Escuela;

public class UbicacionViewModel extends AndroidViewModel {

    private String url = getApplication().getResources().getString(R.string.BASE_URL);
    private MutableLiveData<Escuela> escuela;
    private MutableLiveData<Boolean> estado = new MutableLiveData<>();

    public UbicacionViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<Escuela> getEscuela(){
        EnviarPeticion();
        return escuela;
    }

    public LiveData<Boolean> getEstado(){
        return estado;
    }

    private void EnviarPeticion() {

        escuela = new MutableLiveData<>();

        PeticionesApi peticionesApi = RetrofitClient.getRetrofitInstance().create(PeticionesApi.class);
        peticionesApi.getEscuela(1)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Observer<Escuela>() {
                    @Override
                    public void onSubscribe(Disposable d) {
                        estado.setValue(true);
                    }
                    @Override
                    public void onNext(Escuela response) {
                        escuela.setValue(response);
                    }
                    @Override
                    public void onError(Throwable e) {
                        escuela.setValue(null);
                        estado.setValue(false);
                    }
                    @Override
                    public void onComplete() {
                        estado.setValue(false);
                    }
                });
    }
}

File: presentation/src/main/java/co/edu/uniautonoma/posgradosapp/presentation/ui/ubicacion/UbicacionViewModel.kt
package co.edu.uniautonoma.posgradosapp.presentation.ui.ubicacion

import androidx.lifecycle.*
import co.edu.uniautonoma.posgradosapp.domain.entity.Escuela
import co.edu.uniautonoma.posgradosapp.domain.usecase.EscuelaUseCase
import co.edu.uniautonoma.posgradosapp.domain.util.Result
import kotlinx.coroutines.launch

class UbicacionViewModel(private val escuelaUseCase: EscuelaUseCase) : ViewModel() {

    private val escuela = MediatorLiveData<Escuela>()
    private val error = MediatorLiveData<String>()

    val escuelaLiveData: LiveData<Escuela> get() = escuela
    val errorLiveData: LiveData<String> get() = error

    private fun getUbicacion() {
        viewModelScope.launch {
            val result = escuelaUseCase.getEscuela()
            when(result){
                is Result.Success -> escuela.postValue(result.data)
                is Result.Error -> error.postValue(result.exception.message)
            }
        }
    }

    init {
        getUbicacion()
    }
}